Cookbook/Technical-Documents/From_0%_to_100%_with_Bento.md

```diff
@@ -0,0 +1,426 @@
+            )
+        )
+    }
+
+    private func renderDeliveryOption() -> Node<NodeID> {
+        return Node(
+            id: .delivery,
+            component: appearance.row3(
+                title: Localization.NHS.Prescriptions.deliveryPharmacyOptionTitle,
+                detail: Localization.NHS.Prescriptions.deliveryPharmacyOptionSubtitle,
+                image: Icon(
+                    template: appearance.tokens.icons.image(.ambulance),
+                    backgroundColor: appearance.tokens.colors.grey50
+                ),
+                didTap: ^Action.didTapPharmacyDelivery >>> observer
+            )
+        )
+    }
+```
+
+The `^` and `>>>` are our operators to simplify our codebase. The `^` is a lift operator which lifts single value `Action.didTapPharmacyCollection` into a function `() -> Action`. Why do we need to do so? The `>>>` composition operator needs 2 functions `(A -> B) >>> (B -> C) = (A -> C)`. In our case it's `(Void -> Action) >>> (Action -> Void) = (Void -> Void)` where `didTap` requires `Void -> Void`. Everything matches! 
+
+</br>
+
+> To find out more on functional operator in our code base check out [this article](https://github.com/babylonhealth/ios-playbook/blob/master/Cookbook/Technical-Documents/FunctionalOperators.md).
+
+</br>
+
+#### Multiple rows
+`---+` operator allows you only to append a single node to a section. Very often we need to display a list of models. Bento has also `---*` operator which serves this purpose. This is how it would like in the code:
+
+```swift
+func renderList(of prescriptions: [Prescription]) -> Box<SectionID, NodeID> {  
+	return .empty
+		|-+ Section(id: .first)
+		|---* prescriptions.map { prescription in
+			return Node(
+				id: .prescription(prescription.id),
+				component: appearance.row1(title: prescription.drugName)
+			)
+}
+     
+enum NodeID {
+    case prescription(id: String) 
+}
+```
+
+## How can I create a fresh new component?
+Sometimes we need to create a new component. Usually, it's because the component hasn't been used by another iOS dev. 
+
+What's the process? First of all, look at the checklist what needs to be done:
+
+### New component checklist
+1. Check if the component is in Zeplin under `Design System` tag. Chase a designer if it's not there :)
+2. If the component is there, you can implement it by adding a factory method as an extension of the ComponentsBuilder `extension DesignLibrary.ComponentsBuilder`
+3. Record snapshot tests
+4. Add a demo of the component into GalleryApp
+
+### Writing a new component
+Here we're going to focus on point 2. We are going to write a banner component:
+![](Assets/bento-image-03.png)
+
+We build components by putting **small blocks together**, usually in a StackView. By small blocks, I mean other components which we call `atomic components`. They are called `atomic` because those components are small, usually display a **small unit** as Label, ImageView, plain UIView.
```

Comment:
```suggestion
We build components by putting **small blocks together**, usually in a StackView. By small blocks, we mean other components which we call `atomic components`. They are called `atomic` because those components are small, usually display a **small unit** as Label, ImageView, plain UIView.
```